### ISSUE TITLE
Extra cart line information lost since WooCommerce 3

### ISSUE THREAD

**Issue by webdados** [woocommerce/woocommerce#15470]
### EXPLANATION OF THE ISSUE

We have developed a small plugin to define discounts based on several rules.
We are using the "woocommerce_before_calculate_totals" filter to calculate the new prices based on those rules.
We're also storing the original value as an extra cart line array entry called "taxonomy_discounts", so the next time the cart is loaded on page, we can check if we already applied the discount instead of calculating it again against the already discounted value now on the cart line.

We set that extra information like this at the end of our function attached to "woocommerce_before_calculate_totals":
`WC()->cart->cart_contents[$cart_item_key]['taxonomy_discounts'] = $our_discount_information_array;`

At the beginning of the same function, we check if this information already exists:
`
if ( isset( $cart->cart_contents[$cart_item_key]['taxonomy_discounts'] ) ) {
//Discount already set, just calculate it again based on the original price that we've stored here and eventually new quantities
} else {
//Do the discount calculation from scratch
}
`

**This all works wonderfully on WooCommerce 2.* but on WooCommerce 3.* that 'taxonomy_discounts' entry that we create on the array disappears somehow** and then the plugin calculates the discount over an already discounted value.

Bottom line: how should additional information be stored on a cart line so it's not lost on WooCommerce 3? Is this a bug, or something has changed?

**Comment by webdados** [comment]
This is not a WooCommerce 3 problem. It was a bug on our site.

**Comment by webdados** [comment]
Please delete the issue if possible.
